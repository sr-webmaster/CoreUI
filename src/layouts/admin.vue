<template>
  <div>
    <slot name="navigation-drawer">
      <v-navigation-drawer
        v-model="isDrawerOpen"
        :clipped="isClipped"
        :mini-variant="typeof isMini === 'boolean' ? isMini : false"
        :dark="typeof isDark === 'boolean' ? isDark : true"
        :color="typeof isDark === 'boolean' ? '' : 'white'"
        :width="typeof hideIcons === 'boolean' && hideIcons === true ? 200 : 300"
        disable-resize-watcher
        stateless
        app
      >
        <f-navigation-drawer-list
          :items="items"
          :no-actions="typeof hideIcons === 'boolean' ? hideIcons : false"
          :background-active-color="backgroundActiveColor"
          :foreground-active-color="foregroundActiveColor"
        />
      </v-navigation-drawer>
    </slot>
    <slot name="navigation-top">
      <v-toolbar
        fixed
        app
        :clipped-left="isClipped"
      >
        <v-toolbar-side-icon @click.stop="isDrawerOpen = !isDrawerOpen" />
        <v-toolbar-title>
          <v-flex>
            <v-img
              contain
              height="38"
              width="178"
              position="left center"
              :src="logo"
            />
          </v-flex>
        </v-toolbar-title>
        <v-spacer />
        <v-menu
          v-show="notifications"
          left
          offset-y
          max-width="320"
          origin="right top"
          :disabled="!notifications.length"
        >
          <v-btn
            slot="activator"
            icon
            flat
            :disabled="!notifications.length"
          >
            <v-badge
              color="red"
              overlap
            >
              <span
                v-if="notifications.length"
                slot="badge"
              >
                {{ notifications.length }}
              </span>
              <v-icon
                medium
                color="grey darken-2"
              >
                notifications
              </v-icon>
            </v-badge>
          </v-btn>

          <f-notification-menu
            v-if="!!notificationsFetchInterval"
            :interval="notificationsFetchInterval"
          />
        </v-menu>
        <v-menu
          v-if="currentUser"
          left
          offset-y
          content-class="user-menu"
          max-width="300"
        >
          <v-btn
            slot="activator"
            icon
            flat
            class="avatar"
          >
            <f-user-avatar
              :user="currentUser"
              :size="34"
              :is-dark="isDark"
            />
          </v-btn>

          <f-user-menu
            :user="currentUser"
            :user-is-admin="isCurrentUserAdmin"
            :menu-items="userMenuItems"
            :consumer-view="isConsumerViewAvailable"
            :user-field="displayedUserField"
            :hide-level="hideUserLevel"
            @signout="signout"
          />
        </v-menu>
      </v-toolbar>
    </slot>
    <v-content>
      <slot>
        <v-layout
          v-if="isLoading"
          column
          align-center
        >
          <v-progress-linear
            class="my-0"
            :indeterminate="true"
            :color="loadingColor"
            :height="loadingHeight"
          />
          <h2 :class="`${loadingColor}-text`">
            Loading ... Please Wait
          </h2>
        </v-layout>
        <router-view v-show="!isLoading" />
      </slot>
    </v-content>
    <slot name="messages">
      <v-snackbar
        v-model="isVisible"
        color="error"
        :timeout="6000"
        top
      >
        {{ errorMessages }}
        <v-btn
          dark
          flat
          @click="setErrorVisibility(false)"
        >
          Close
        </v-btn>
      </v-snackbar>
      <v-snackbar
        v-model="isMessageVisible"
        color="success"
        :timeout="6000"
        top
      >
        {{ message }}
        <v-btn
          dark
          flat
          @click="setMessageVisibility(false)"
        >
          Close
        </v-btn>
      </v-snackbar>
    </slot>
    <slot name="footer">
      <f-footer :color="footerColor" />
    </slot>
  </div>
</template>

<script>
import { mapState, mapGetters, mapActions } from 'vuex'
import { createHelpers } from 'vuex-map-fields'
import FUserAvatar from '@freshinup/core-ui/src/components/FUserAvatar'
import FUserMenu from '@freshinup/core-ui/src/components/FUserMenu'
import FNavigationDrawerList from '@freshinup/core-ui/src/components/FNavigationDrawerList'
import FNotificationMenu from '@freshinup/core-ui/src/components/FNotificationMenu'
import FFooter from '@freshinup/core-ui/src/components/FFooter'

const generalErrorMessageFields = createHelpers({
  getterType: 'generalErrorMessages/getField',
  mutationType: 'generalErrorMessages/updateField'
}).mapFields

const generalMessageFields = createHelpers({
  getterType: 'generalMessage/getField',
  mutationType: 'generalMessage/updateField'
}).mapFields

const navigationAdminFields = createHelpers({
  getterType: 'navigationAdmin/getField',
  mutationType: 'navigationAdmin/updateField'
}).mapFields

export default {
  components: {
    FUserMenu,
    FNavigationDrawerList,
    FNotificationMenu,
    FFooter,
    FUserAvatar
  },
  computed: {
    ...generalErrorMessageFields([
      'isVisible'
    ]),
    ...navigationAdminFields({
      isDrawerOpen: 'isDrawerOpen',
      isClipped: 'isClipped'
    }),
    ...generalMessageFields({
      isMessageVisible: 'isVisible'
    }),
    ...mapState('navigationAdmin', [
      'items',
      'logo',
      'isDark',
      'isMini',
      'hideIcons',
      'backgroundActiveColor',
      'foregroundActiveColor',
      'footerColor'
    ]),
    ...mapGetters('generalErrorMessages', {
      errorMessages: 'errorMessages'
    }),
    ...mapGetters('generalMessage', {
      message: 'message'
    }),
    ...mapGetters(['currentUser']),
    ...mapGetters('currentUser', {
      isCurrentUserAdmin: 'isAdmin'
    }),
    ...mapState('navigation', {
      userMenuItems: state => state.userMenuItems,
      isConsumerViewAvailable: state => state.isConsumerViewAvailable,
      hideUserLevel: state => state.hideUserLevel,
      displayedUserField: state => state.displayedUserField
    }),
    ...mapGetters(['currentUser']),
    ...mapGetters('userNotifications', {
      notifications: 'unacknowledged'
    }),
    ...mapState('userNotifications', {
      notificationsFetchInterval: 'fetchInterval'
    }),
    ...mapState('page', [
      'isLoading',
      'loadingColor',
      'loadingHeight'
    ])
  },
  methods: {
    ...mapActions('generalErrorMessages', {
      setErrorVisibility: 'setVisibility'
    }),
    ...mapActions('generalMessage', {
      setMessageVisibility: 'setVisibility'
    }),
    signout () {
      this.$store.dispatch('currentUser/logout', { $auth: this.$auth })
    }
  },
  async beforeRouteEnterOrUpdate (vm, to, from, next) {
    try {
      await vm.$store.dispatch('currentUser/getCurrentUser')
    } finally {
      if (next) next()
    }
  }
}
</script>
