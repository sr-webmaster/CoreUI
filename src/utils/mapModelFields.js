import get from 'lodash/get'
import isEqual from 'lodash/isEqual'
import capitalize from 'lodash/capitalize'

export default (fields = [], model = { prop: 'value', event: 'input' }) => {
  const result = {}
  fields.forEach((field) => {
    result[field] = {
      get () { return get(this[model.prop], field) },
      set (val) {
        const currentValue = get(this[model.prop], field)
        const emittingMethod = this['emit' + capitalize(model.event)]
        if (!isEqual(val, currentValue) && emittingMethod) {
          emittingMethod({ [field]: val })
        }
      }
    }
  })
  return result
}
