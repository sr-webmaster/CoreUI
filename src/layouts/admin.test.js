import { shallowMount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import createStore from 'tests/createStore'
import { mount } from 'vue-cli-plugin-freshinup-ui/utils/testing'
import Component from '@freshinup/core-ui/src/layouts/admin.vue'
import * as Stories from './admin.stories'

describe('Admin layout', () => {
  let localVue, store
  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      store = createStore({
        page: {
          isLoading: false
        }
      })
    })
    test('signout()', () => {
      const dispatchMock = jest.fn()
      const authMock = jest.fn()
      store.dispatch = dispatchMock
      const wrapper = shallowMount(Component, {
        localVue,
        store,
        mocks: {
          $auth: authMock
        }
      })
      const vm = wrapper.vm
      vm.signout()
      expect(store.dispatch).toHaveBeenCalledWith('currentUser/logout', { $auth: authMock })
    })
  })
  describe('Visuals', () => {
    test('with current user', (done) => {
      const wrapper = mount(Stories.UserWithAvatar(), { pluginRouter: true })
      expect(wrapper.isVueInstance()).toBe(true)
      wrapper.beforeRouteEnterOrUpdate({}, null, async () => {
        await wrapper.vm.$nextTick()
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
    test('Error Message', async (done) => {
      const wrapper = mount(Stories.ErrorMessage(), { pluginRouter: true })
      await wrapper.beforeRouteEnterOrUpdate({}, {}, async () => {
        wrapper.vm.$store.dispatch('generalErrorMessages/setErrors', 'User name must be set before saving')
        wrapper.vm.$store.dispatch('generalErrorMessages/setVisibility', true)
        await wrapper.vm.$nextTick()
        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('User name must be set before saving')
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
  })
})
