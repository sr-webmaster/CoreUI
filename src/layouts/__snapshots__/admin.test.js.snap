// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin layout Visuals Error Message 1`] = `
<div>
  <aside
    class="v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--open theme--dark"
    color="white"
    style="height: 100%; margin-top: 0px; max-height: calc(100% - 32px); transform: translateX(0px); width: 300px;"
  >
    <div
      class="v-list theme--dark"
      role="list"
    >
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              dashboard
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Dashboard
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="primary--text v-list__tile--active v-list__tile v-list__tile--link theme--dark"
              href="#/"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Home
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              supervised_user_circle
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Base User
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/users"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Users
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/teams"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Teams
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/companies"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Companies
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              settings
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Platform Settings
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="primary--text v-list__tile--active v-list__tile v-list__tile--link theme--dark"
              href="#/"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Terms
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
   
  <nav
    class="v-toolbar v-toolbar--fixed theme--light"
    style="margin-top: 0px; padding-right: 0px; padding-left: 300px; transform: translateY(0px);"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-toolbar__side-icon v-btn v-btn--icon theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            menu
          </i>
        </div>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        <div
          class="flex"
        >
          <div
            class="v-responsive v-image"
            style="height: 38px; width: 178px;"
          >
            <div
              class="v-image__image v-image__image--preload v-image__image--contain"
              style="background-position: left center;"
            />
            <div
              class="v-responsive__content"
            />
          </div>
        </div>
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="v-menu v-menu--inline"
      >
        <div
          class="v-menu__activator v-menu__activator--disabled"
        >
          <button
            class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
            disabled="disabled"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <span
                class="v-badge v-badge--overlap"
              >
                <!---->
                 
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light grey--text text--darken-2"
                  style="font-size: 28px;"
                >
                  notifications
                </i>
                <!---->
              </span>
            </div>
          </button>
        </div>
        <div
          class="v-menu__content theme--light "
          style="max-height: auto; min-width: 0px; max-width: 320px; top: 12px; left: 12px; transform-origin: right top; z-index: 0; display: none;"
        />
      </div>
       
      <div
        class="v-menu v-menu--inline"
      >
        <div
          class="v-menu__activator"
        >
          <button
            class="avatar v-btn v-btn--flat v-btn--icon theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <div
                class="v-avatar accent"
                style="height: 34px; width: 34px;"
              >
                <span
                  class="accent-text--text"
                  style="font-size: 17px;"
                >
                  
    DA
  
                </span>
              </div>
            </div>
          </button>
        </div>
        <div
          class="v-menu__content theme--light user-menu"
          style="max-height: auto; min-width: 0px; max-width: 300px; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
           
          <div
            class="v-card v-sheet theme--light"
          >
            <div
              class="layout pa-3 row"
            >
              <div
                class="flex shrink"
              >
                <div
                  class="v-avatar accent"
                  style="height: 64px; width: 64px;"
                >
                  <span
                    class="accent-text--text"
                    style="font-size: 32px;"
                  >
                    
    DA
  
                  </span>
                </div>
              </div>
               
              <div
                class="flex pl-3"
              >
                <h4
                  class="title secondary--text"
                >
                  
        Demo Admin
      
                </h4>
                 
                <span
                  class="v-chip mx-0 pl-3 pr-3 f-userMenu__userLevel v-chip--disabled v-chip--small theme--light secondary white--text"
                  tabindex="-1"
                >
                  <span
                    class="v-chip__content"
                  >
                    
        Platform Admin
      
                  </span>
                </span>
                 
                <div>
                  <!---->
                  
          Laravel
        
                </div>
              </div>
            </div>
             
            <hr
              class="v-divider theme--light"
            />
             
            <div
              class="v-list pb-0 theme--light"
              role="list"
            >
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Profile
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Teams
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Company
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Settings
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Company Settings
                    </div>
                  </div>
                </a>
              </div>
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                  href="/admin"
                >
                  
        Switch to Admin View
      
                </a>
              </div>
               
              <hr
                class="v-divider theme--light"
              />
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Sign Out
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
   
  <main
    class="v-content"
    style="padding: 64px 0px 32px 300px;"
  >
    <div
      class="v-content__wrap"
    />
  </main>
   
  <div
    class="v-snack v-snack--active v-snack--top v-snack-transition-enter v-snack-transition-enter-active"
  >
    <div
      class="v-snack__wrapper error"
    >
      <div
        class="v-snack__content"
      >
        
      User name must be set before saving 
      
        <button
          class="v-btn v-btn--flat theme--dark"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
        Close
      
          </div>
        </button>
      </div>
    </div>
  </div>
   
  <!---->
   
  <footer
    class="v-footer px-3 white--text v-footer--fixed theme--light accent"
    style="height: 32px;"
  >
    <div
      class="layout row wrap"
    >
      <span>
        Powered by FreshinUp
      </span>
       
      <div
        class="spacer"
      />
       
      <span>
        © 2020
      </span>
    </div>
  </footer>
</div>
`;

exports[`Admin layout Visuals with current user 1`] = `
<div>
  <aside
    class="v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--open theme--dark"
    color="white"
    style="height: 100%; margin-top: 0px; max-height: calc(100% - 32px); transform: translateX(0px); width: 300px;"
  >
    <div
      class="v-list theme--dark"
      role="list"
    >
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              dashboard
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Dashboard
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="primary--text v-list__tile--active v-list__tile v-list__tile--link theme--dark"
              href="#/"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Home
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              supervised_user_circle
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Base User
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/users"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Users
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/teams"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Teams
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
          <div
            role="listitem"
          >
            <a
              class="v-list__tile v-list__tile--link theme--dark"
              href="#/admin/companies"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Companies
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="v-list__group"
      >
        <div
          class="v-list__group__header"
        >
          <div
            class="v-list__group__header__prepend-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              settings
            </i>
          </div>
          <div
            role="listitem"
          >
            <div
              class="v-list__tile theme--dark"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Platform Settings
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-list__group__header__append-icon"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--dark"
            >
              keyboard_arrow_down
            </i>
          </div>
        </div>
        <div
          class="v-list__group__items v-list__group__items--no-action"
          style="display: none;"
        >
           
          <div
            role="listitem"
          >
            <a
              class="primary--text v-list__tile--active v-list__tile v-list__tile--link theme--dark"
              href="#/"
            >
              <div
                class="v-list__tile__content"
              >
                <div
                  class="v-list__tile__title"
                >
                  Terms
                </div>
              </div>
               
              <div
                class="v-list__tile__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--dark"
                />
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
   
  <nav
    class="v-toolbar v-toolbar--fixed theme--light"
    style="margin-top: 0px; padding-right: 0px; padding-left: 300px; transform: translateY(0px);"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-toolbar__side-icon v-btn v-btn--icon theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            menu
          </i>
        </div>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        <div
          class="flex"
        >
          <div
            class="v-responsive v-image"
            style="height: 38px; width: 178px;"
          >
            <div
              class="v-image__image v-image__image--preload v-image__image--contain"
              style="background-position: left center;"
            />
            <div
              class="v-responsive__content"
            />
          </div>
        </div>
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="v-menu v-menu--inline"
      >
        <div
          class="v-menu__activator v-menu__activator--disabled"
        >
          <button
            class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
            disabled="disabled"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <span
                class="v-badge v-badge--overlap"
              >
                <!---->
                 
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light grey--text text--darken-2"
                  style="font-size: 28px;"
                >
                  notifications
                </i>
                <!---->
              </span>
            </div>
          </button>
        </div>
        <div
          class="v-menu__content theme--light "
          style="max-height: auto; min-width: 0px; max-width: 320px; top: 12px; left: 12px; transform-origin: right top; z-index: 0; display: none;"
        />
      </div>
       
      <div
        class="v-menu v-menu--inline"
      >
        <div
          class="v-menu__activator"
        >
          <button
            class="avatar v-btn v-btn--flat v-btn--icon theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <div
                class="v-avatar accent"
                style="height: 34px; width: 34px;"
              >
                <img
                  alt="Demo Admin"
                  src="https://i.pravatar.cc/300"
                />
              </div>
            </div>
          </button>
        </div>
        <div
          class="v-menu__content theme--light user-menu"
          style="max-height: auto; min-width: 0px; max-width: 300px; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
           
          <div
            class="v-card v-sheet theme--light"
          >
            <div
              class="layout pa-3 row"
            >
              <div
                class="flex shrink"
              >
                <div
                  class="v-avatar accent"
                  style="height: 64px; width: 64px;"
                >
                  <img
                    alt="Demo Admin"
                    src="https://i.pravatar.cc/300"
                  />
                </div>
              </div>
               
              <div
                class="flex pl-3"
              >
                <h4
                  class="title secondary--text"
                >
                  
        Demo Admin
      
                </h4>
                 
                <span
                  class="v-chip mx-0 pl-3 pr-3 f-userMenu__userLevel v-chip--disabled v-chip--small theme--light secondary white--text"
                  tabindex="-1"
                >
                  <span
                    class="v-chip__content"
                  >
                    
        Platform Admin
      
                  </span>
                </span>
                 
                <div>
                  <!---->
                  
          Laravel
        
                </div>
              </div>
            </div>
             
            <hr
              class="v-divider theme--light"
            />
             
            <div
              class="v-list pb-0 theme--light"
              role="list"
            >
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Profile
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Teams
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Company
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Settings
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="secondary--text darken-3"
                role="listitem"
              >
                <a
                  class="primary white--text v-list__tile--active v-list__tile v-list__tile--link theme--light"
                  href="#/"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      My Company Settings
                    </div>
                  </div>
                </a>
              </div>
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                  href="/admin"
                >
                  
        Switch to Admin View
      
                </a>
              </div>
               
              <hr
                class="v-divider theme--light"
              />
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Sign Out
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
   
  <main
    class="v-content"
    style="padding: 64px 0px 32px 300px;"
  >
    <div
      class="v-content__wrap"
    />
  </main>
   
  <!---->
   
  <!---->
   
  <footer
    class="v-footer px-3 white--text v-footer--fixed theme--light accent"
    style="height: 32px;"
  >
    <div
      class="layout row wrap"
    >
      <span>
        Powered by FreshinUp
      </span>
       
      <div
        class="spacer"
      />
       
      <span>
        © 2020
      </span>
    </div>
  </footer>
</div>
`;
